src/vma/dev/allocator.cpp

```diff
@@ -125,14 +125,16 @@ void* vma_allocator::alloc_and_reg_mr(size_t size, ib_ctx_handler *p_ib_ctx_h)
 	default:
 		__log_info_dbg("allocating memory using malloc()");
 		align_simple_malloc(size); // if fail will raise exception
-		m_mem_alloc_type = ALLOC_TYPE_ANON;
 		if (!register_memory(size, p_ib_ctx_h, access)) {
 			__log_info_dbg("failed registering data memory block");
 			throw_vma_exception("failed registering data memory block");
 		}
+		m_mem_alloc_type = ALLOC_TYPE_ANON;
```

Comment:
this should stay in line 128, if registration fails the cleanup won't work properly

Reply:
I think that setting `m_mem_alloc_type = ALLOC_TYPE_ANON;` should be done after we see that allocation/registration are done successfully.
According current implementation VMA catches exception in case failure during register_memory().